src/searcher_frame_document.rs

```diff
@@ -0,0 +1,232 @@
+#![allow(clippy::new_ret_no_self)]
+
+use futures::StreamExt;
+use std::collections::BTreeSet;
+use std::iter::FromIterator;
+use std::os::linux::raw::stat;
```

Comment:
This is deprecated, it says to use this https://docs.rs/libc/latest/libc/struct.stat.html